chainstate_test_framework/src/lib.rs

```diff
@@ -22,9 +22,13 @@ mod transaction_builder;
 pub type TestStore = chainstate_storage::inmemory::Store;
 
 /// Chainstate instantiation for testing, using the in-memory storage backend
-pub type TestChainstate = crate::Chainstate<TestStore>;
+pub type TestChainstate = Box<dyn chainstate::chainstate_interface::ChainstateInterface>;
 
-pub use self::{
-    block_builder::BlockBuilder, framework::TestFramework, framework_builder::TestFrameworkBuilder,
+pub use {
+    block_builder::BlockBuilder, framework::anyonecanspend_address, framework::empty_witness,
+    framework::TestBlockInfo, framework::TestFramework, framework_builder::TestFrameworkBuilder,
     transaction_builder::TransactionBuilder,
 };
+
+#[cfg(test)]
+mod tests;
```

Comment:
I feel like these tests belong somewhere else. I would expect for the tests in the `test_framework` crate to test the framework itself and these are chainstate tests. I think it would be better to move them to a separate crate.

Reply:
I'll see what I can do.